src/pages/home/report/ReportActionsView.tsx

```diff
@@ -49,34 +51,72 @@ type ReportActionsViewProps = ReportActionsViewOnyxProps & {
 
     /** The report actions are loading newer data */
     isLoadingNewerReportActions?: boolean;
+
+    /** Whether the report is ready for comment linking */
+    isReadyForCommentLinking?: boolean;
 };
 
+const DIFF_BETWEEN_SCREEN_HEIGHT_AND_LIST = 120;
+const SPACER = 16;
+
 function ReportActionsView({
     report,
     session,
     parentReportAction,
-    reportActions = [],
+    reportActions: allReportActions = [],
     isLoadingInitialReportActions = false,
     isLoadingOlderReportActions = false,
     isLoadingNewerReportActions = false,
+    isReadyForCommentLinking = false,
 }: ReportActionsViewProps) {
     useCopySelectionHelper();
     const reactionListRef = useContext(ReactionListContext);
+    const route = useRoute<RouteProp<CentralPaneNavigatorParamList, typeof SCREENS.REPORT>>();
+    const reportActionID = route?.params?.reportActionID;
     const didLayout = useRef(false);
     const didSubscribeToReportTypingEvents = useRef(false);
-    const isFirstRender = useRef(true);
-    const hasCachedActions = useInitialValue(() => reportActions.length > 0);
-    const mostRecentIOUReportActionID = useMemo(() => ReportActionsUtils.getMostRecentIOURequestActionID(reportActions), [reportActions]);
+
     const network = useNetwork();
     const {isSmallScreenWidth} = useWindowDimensions();
+    const contentListHeight = useRef(0);
+    const layoutListHeight = useRef(0);
+    const {windowHeight} = useWindowDimensions();
+    const isFocused = useIsFocused();
     const prevNetworkRef = useRef(network);
     const prevAuthTokenType = usePrevious(session?.authTokenType);
-
+    const [isInitialLinkedView, setIsInitialLinkedView] = useState(!!reportActionID);
```

Comment:
```suggestion
    const [isLinkedActionInList, setIsLinkedActionInList] = useState(!!reportActionID);
```

Reply:
Also, should this be initialized to false if a reportActionID is provided but we don't know if it's in the list yet?

```suggestion
    const [isInitialLinkedView, setIsInitialLinkedView] = useState(false);
```

Reply:
This prop indicates that we are currently navigating to the linked message. We made it to work with the autoscroll feature `autoscrollToTopThreshold` in the list — it needs to be true when we've got a reportActionID so we don't end up scrolling when we shouldn't. I can rename it 
`isNavigatingToLinkedMessage`
`isLinkingToMessageInitiated`
`isMessageLinkNavigationInProgress`
I would stay with the first one